### ISSUE TITLE
Opensuse leap 15.4 certificate `cert.pem` is invalid. Broken link `http://bit.ly/ruby-ssl`

### ISSUE THREAD

**Issue by KarlHeitmann** [rbenv/rbenv#1425]
Hello!

I'm running opensuse leap 15.4, and tried to install rbenv to manage my ruby versions. Once upon I installed ruby 2.7.2, when I tried to `bundle install` the gems of my project, I've got this error `Could not verify the SSL certificate for https://rubygems.org/`, and tried to reach the link `http://bit.ly/ruby-ssl` the link was broken. Opensuse leap 15.4 has currently `openssl 1.1.1` in its official repository.

Before installing ruby 2.7.2, I compiled `https://github.com/openssl/openssl/releases/tag/OpenSSL_1_1_1q` with these flags: `./Configure --prefix=/opt/openssl_1_1_1q --openssldir=/opt/openssl_1_1_1q/ssl linux-x86_64` to install it in an isolated directory. Then I set the environment variable `RUBY_CONFIGURE_OPTS=--with-openssl-dir=/opt/openssl_1_1_1q/` and finally `rbenv install 2.7.2`

After this build, I ran the command `bundle install` and stumble with the problem described on the first paragraph. Here is a screenshot:

![image](https://user-images.githubusercontent.com/3003032/188786192-96b1be90-f913-497a-b48d-1a3404552678.png)

I followed[ this guide](https://bundler.io/guides/rubygems_tls_ssl_troubleshooting_guide.html), but I broke so many things in the process, even broke my system ruby when running `gem update --system` (with sudo) so I had to reinstall my OS.

After some tinkering, I finally found out that the `cert.pem` certificate was outdated. So I took this certificate: `https://raw.githubusercontent.com/rubygems/rubygems/master/lib/rubygems/ssl_certs/rubygems.org/GlobalSignRootCA_R3.pem` and save it under the file in `/opt/openssl_1_1_1q/ssl/cert.pem`. I discovered that was the right name and place for the file because after running the command `curl -Lks 'https://git.io/rg-ssl' | ruby` mentiond in the [guide above](https://bundler.io/guides/rubygems_tls_ssl_troubleshooting_guide.html#automated-ssl-check), I've got this reply from the terminal:

![image](https://user-images.githubusercontent.com/3003032/188787038-fee45213-b06e-44b6-b224-384574b17e1f.png)
:warning: :warning: :warning: :warning:  the screenshot has the route `/opt/openssl_1_1_1d/ssl/cert.pem` , but it should be `/opt/openssl_1_1_1q/ssl/cert.pem`, the screenshot is from another test I made. But it's the same process. :warning: :warning: :warning: :warning: 

Once the file is created, I could connect to rubygems.

**Comment by mislav** [comment]
Thanks for sharing your problem and workaround!

Closing because rbenv itself doesn't deal with building Ruby nor running RubyGems. Therefore, this is a problem with how the OpenSSL installation itself is managed.

**Comment by KarlHeitmann** [comment]
Hi @mislav , I understand the issue is off-topic here. Maybe I should post it on the `ruby-build` plugin repo?

Just wanted to give the heads-up, because this year I've had to change my laptop, and make a fresh install of rbenv in Ubuntu 22.04 and Open Suse Leap 15.4, in both OS I've had issues with openssl, I think it is because on Ubuntu they started using openssl 3.x on the official repository. IMHO a bad decision, because it is breaking a lot of stuff. That's the reason why I then switched back to OpenSuse that uses more stable software.
